app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -89,6 +89,21 @@ class UserViewModel(
         })
   }
 
+  /**
+   * Checks if a user exists in the database.
+   *
+   * @param user the user to check
+   * @param onSuccess the callback to call with a boolean indicating if the user exists
+   */
+  fun doesUserExist(user: User, onSuccess: (Boolean) -> Unit) {
+    userRepository.userExists(
+        user,
+        onSuccess = onSuccess,
+        onFailure = {
+          Log.e("UserViewModel", "Failed to check if user exists: ${user.public.userId}", it)
```

Comment:
I don't know if it is a log of debug and if it is intended to stay. But if it is, wouldn't it be more useful if they were both in the `userExist()` function in the repository so that we are sure that there is a problem with the repo if need be ?

Reply:
Right, I moved the `Log` down to the `UserRepository` for consistency but left the `onFailure` callback in case it will be needed later.